ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedWindowTableIQv2Test.java

```diff
@@ -0,0 +1,306 @@
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  @Test
+  public void shouldThrowNPEs() {
+    new NullPointerTester()
+        .setDefault(KsStateStore.class, stateStore)
+        .testConstructors(KsMaterializedWindowTable.class, Visibility.PACKAGE);
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails() {
+    // Given:
+    when(stateStore.getKafkaStreams()).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails_fetchAll() {
+    // Given:
+    when(stateStore.getKafkaStreams()).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to scan materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfStoreFetchFails() {
+    // Given:
+    when(kafkaStreams.query(any()))
+        .thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfStoreFetchFails_fetchAll() {
+    // Given:
+    when(kafkaStreams.query(any()))
+      .thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to scan materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldReturnValuesForClosedStartBounds() {
+    // Given:
+    final Range<Instant> start = Range.closed(
+      NOW,
+      NOW.plusSeconds(10)
+    );
+
+    when(kafkaStreams.query(any())).thenReturn(queryResult);
+    when(queryResult.getGlobalResult()).thenReturn(fetchResult);
+    when(fetchResult.getResult()).thenReturn(fetchIterator);
+    when(fetchIterator.hasNext()).thenReturn(true, true, false);
```

Comment:
Similar concern to https://github.com/confluentinc/ksql/pull/8556#discussion_r779933662

Reply:
I changed this one to not use the globalresult